File: cloud/routes/sync.js
const express = require("express");

const { verificarToken, requierePermiso } = require("../middlewares/auth");
// const { DiaProgramacion } = require("../models");
// const { where } = require("sequelize");


module.exports = () => {
    
    const router = express.Router();
    const syncController = require("../controllers/syncController")();

    // GET /api/sync/diasemana
    router.get('/diasemana', syncController.dia_semana);

    // GET /api/sync/programaciones
    router.get('/programaciones', syncController.programaciones);

    // GET /api/sync/dias-programados
    router.get('/dias-programados', syncController.dias_programados);

    // GET /api/sync/sectores
    router.get('/sectores', syncController.sectores);

    // GET /api/sync/placas
    router.get('/placas', syncController.placas);

    // GET /api/sync/valvulas
    router.get('/valvulas', syncController.valvulas);

    // GET /api/sync/sensores
    router.get('/sensores', syncController.sensores);

    return router;
}

File: raspberry/database/sync.py
import requests # type: ignore
from database.db_helper import DBHelper
from config.config import BASE_URL

BASE_URL = f"{BASE_URL}/api/sync"

def sincronizar_todo():
    endpoints = {
        "diasemana": DBHelper.sincronizar_dia_semana,
        "programaciones": DBHelper.sincronizar_programacion_horario,
        "dias-programados": DBHelper.sincronizar_dia_programacion,
        "sectores": DBHelper.sincronizar_sector,
        "placas": DBHelper.sincronizar_placa,
        "valvulas": DBHelper.sincronizar_valvula,
        "sensores": DBHelper.sincronizar_sensor_flujo,
    }

    for nombre, funcion in endpoints.items():
        try:
            res = requests.get(f"{BASE_URL}/{nombre}")
            if res.status_code == 200:
                data = res.json()
                funcion(data)
                print(f"✅ {nombre} sincronizado")
            else:
                print(f"Error al obtener {nombre}")
        except Exception as e:
            print(f"Error en {nombre}: {e}")
